src/npc.cpp

```diff
@@ -385,9 +386,20 @@ void npc::randomize( const npc_class_id &type )
     starting_inv( *this, type );
     has_new_items = true;
 
-    for( const auto &pr : type->traits ) {
-        if( rng( 1, 100 ) <= pr.second ) {
-            set_mutation( pr.first );
+    my_mutations.clear();
+    my_traits.clear();
+
+    // Add fixed traits
+    Trait_list tmplist = trait_group::traits_from( type->traits );
```

Comment:
You don't nee this variable, you can iterate over the result of the function cal directly `for( ... : traits_from() )`.